sycl/include/sycl/ext/oneapi/matrix/matrix-jit-use.hpp

```diff
@@ -0,0 +1,660 @@
+
+template <layout Layout> struct spv_matrix_layout_traits {
+  static constexpr __spv::MatrixLayout value = __spv::MatrixLayout::Unused;
+};
+
+#define SPV_MATRIX_LAYOUT_TRAITS(LAYOUT, SPV_LAYOUT)                           \
+  template <> struct spv_matrix_layout_traits<LAYOUT> {                        \
+    static constexpr __spv::MatrixLayout value = SPV_LAYOUT;                   \
+  };
+
+SPV_MATRIX_LAYOUT_TRAITS(layout::row_major, __spv::MatrixLayout::RowMajor)
+SPV_MATRIX_LAYOUT_TRAITS(layout::col_major, __spv::MatrixLayout::ColumnMajor)
+SPV_MATRIX_LAYOUT_TRAITS(layout::packed_a, __spv::MatrixLayout::PackedA)
+SPV_MATRIX_LAYOUT_TRAITS(layout::packed_b, __spv::MatrixLayout::PackedB)
+SPV_MATRIX_LAYOUT_TRAITS(layout::unused, __spv::MatrixLayout::Unused)
+
+// unnecessary was introduced for backward compatibility.
+// Once the use implementation is stable, "unnecessary" value will be omitted
+enum class use { a, b, accumulator, unnecessary };
+
+template <use Use> struct spv_matrix_use_traits {
+  static constexpr __spv::MatrixUse value = __spv::MatrixUse::MatrixA;
+};
+
+#define SPV_MATRIX_USE_TRAITS(USE, SPV_USE)                                    \
+  template <> struct spv_matrix_use_traits<USE> {                              \
+    static constexpr __spv::MatrixUse value = SPV_USE;                         \
+  };
+
+SPV_MATRIX_USE_TRAITS(use::a, __spv::MatrixUse::MatrixA)
+SPV_MATRIX_USE_TRAITS(use::b, __spv::MatrixUse::MatrixB)
+SPV_MATRIX_USE_TRAITS(use::accumulator, __spv::MatrixUse::Accumulator)
+SPV_MATRIX_USE_TRAITS(use::unnecessary, __spv::MatrixUse::Unnecessary)
+
+template <typename G> struct spv_scope_traits {};
+template <> struct spv_scope_traits<sycl::sub_group> {
+  constexpr static auto value = __spv::Scope::Subgroup;
+};
+template <int D> struct spv_scope_traits<sycl::group<D>> {
+  constexpr static auto value = __spv::Scope::Workgroup;
+};
+
+template <typename T, size_t NumRows, size_t NumCols, use Use,
+          layout Layout = layout::unused, typename Group = sycl::sub_group>
+class wi_data;
+template <typename T, size_t NumRows, size_t NumCols, use Use,
+          layout Layout = layout::unused, typename Group = sycl::sub_group>
+struct joint_matrix {
+public:
+  __spv::__spirv_JointMatrixINTEL<
+      T, NumRows, NumCols, spv_matrix_layout_traits<Layout>::value,
+      spv_scope_traits<Group>::value, spv_matrix_use_traits<Use>::value> *spvm;
+  joint_matrix(Group sg) {
+#ifndef __SYCL_DEVICE_ONLY__
+    (void)sg;
+    throw runtime_error("joint matrix is not supported on host device.",
+                        PI_ERROR_INVALID_DEVICE);
+#endif // __SYCL_DEVICE_ONLY__
+  }
+
+  inline __SYCL_ALWAYS_INLINE wi_data<T, NumRows, NumCols, Use, Layout, Group>
+  get_wi_data() {
+    return wi_data<T, NumRows, NumCols, Use, Layout, Group>(*this);
+  }
+};
+
+template <typename Group, typename T, size_t NumRows, size_t NumCols, use Use,
+          layout Layout = layout::unused, access::address_space Space>
```

Comment:
I don't think that this needs `layout Layout = layout::unused` here right? This template paramater should be safely removed since it is not used in intel/cuda/hip for this interface.